### PR TITLE
live suite for cdn testing

### PR DESCRIPTION
**Pull request by sunilangadi2**
Signed-off-by: Sunil Angadi <sangadi@redhat.com>

# Description

Suite for live testing - customer scenario which covers both rpm and container.

# Checklist:

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Comment by sunilangadi2** [comment]
Successful logs:
rhel 7 rpm: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611127078771/result.html
rhel 8 rpm: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611133604052/result.html

rhel 7 container: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611120688716/result.html
rhel 8 container: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611133632721/result.html

**Comment by rakeshgm** [comment]
nice addition.

File: suites/nautilus/live/live_sanity_container.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    config:
+          io-total: 100M
+    desc: Perform export during read/write,resizing,flattening,lock operations
+
+- test:
+    name: rgw sanity tests
+    desc: User and container access in same and different tenants
+    module: sanity_rgw.py
+    config:
+          script-name: test_multitenant_user_access.py
```
Comment (rakeshgm) [inline suites/nautilus/live/live_sanity_container.yaml:86]: use of this [test_Mbuckets_with_Nobjects](https://github.com/red-hat-storage/cephci/blob/master/suites/nautilus/rgw/sanity_rgw.yaml#L61) would be better here than test_multitenant_user_access

File: suites/nautilus/live/live_sanity_rpm.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    config:
+      io-total: 100M
+    desc: Perform export during read/write,resizing,flattening,lock operations
+
+- test:
+    name: rgw sanity tests
+    desc: User and container access in same and different tenants
+    module: sanity_rgw.py
+    config:
+      script-name: test_multitenant_user_access.py
```
Comment (rakeshgm) [inline suites/nautilus/live/live_sanity_rpm.yaml:84]: same as the above comment

**Comment by rakeshgm** [comment]
can you add the Polarion link here. and also add the script link in polarion. 
and don't add PR number in it. add the location of suite and in the next line add the mention the filename.

File: suites/nautilus/live/live_sanity_container.yaml (hunk 1 above)
Comment (sunilangadi2) [reply suites/nautilus/live/live_sanity_container.yaml:86]: Sure, will change that.

File: suites/nautilus/live/live_sanity_rpm.yaml (hunk 2 above)
Comment (sunilangadi2) [reply suites/nautilus/live/live_sanity_rpm.yaml:84]: will change here as well.

**Comment by VasishtaShastry** [comment]
I think we are not doing live testing here
We are pointing vms to rhsm.stage itself

Though these suites are checking ceph-ansible's functionality with latest package in  stage, I think it is not serving the purpose of live testing
We need to find a way to fetch packages from cdn live

**Comment by veera-raghava-reddy** [comment]
@sunilangadi2 can you review and address comments from @VasishtaShastry ?

**Comment by sunilangadi2** [comment]
@veera-raghava-reddy Sure.

File: suites/nautilus/live/live_sanity_container.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            global:
+              osd_pool_default_pg_num: 64
+              osd_default_pool_size: 2
+              osd_pool_default_pgp_num: 64
+              mon_max_pg_per_osd: 1024
+            mon:
+              mon_allow_pool_delete: true
+            client:
+              rgw crypt require ssl: false
+              rgw crypt s3 kms encryption keys: testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo=
```
Comment (psathyan) [inline suites/nautilus/live/live_sanity_container.yaml:47]: Could you bring `testkey-1` into the next line?

File: suites/nautilus/live/live_sanity_container.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    module: install_prereq.py
+    abort-on-fail: True
+
+- test:
+    name: ceph ansible install containerized rhcs 4.x from cdn
+    polarion-id: CEPH-83573588
+    module: test_ansible.py
+    config:
+      use_cdn: True
+      build: '4.x'
```
Comment (psathyan) [inline suites/nautilus/live/live_sanity_container.yaml:13]: Do we need the key `build`? I don't to have a reason to indicate it is not required but it is kind of misleading when we say `4.x`...

File: suites/nautilus/live/live_sanity_container.yaml (hunk 3 above)
Comment (sunilangadi2) [reply suites/nautilus/live/live_sanity_container.yaml:47]: ok sure

File: suites/nautilus/live/live_sanity_container.yaml (hunk 4 above)
Comment (sunilangadi2) [reply suites/nautilus/live/live_sanity_container.yaml:13]: ok will remove that config option key **build**, as i was using earlier yaml hence it got retained.
Thanks

**Comment by veera-raghava-reddy** [comment]
Any update on this PR?

**Comment by sunilangadi2** [comment]
@veera-raghava-reddy This PR is on hold due to this PR https://github.com/red-hat-storage/cephci/pull/339 as we require separate credential for stage and cdn once it got merged I will rework on it.

**Comment by VasishtaShastry** [comment]
@sunilangadi2 
I'm not sure why this PR is on hold because of PR 339
This PR needs cdn credentials and the provision to set cdn credentials are already in the cephci.yaml.template
The PR is actually blocked on procuring live credentials where we actually consume live contents with pool-id with subscription available for testing. The discussion to procure the same is going on.

**Comment by sunilangadi2** [comment]
> @sunilangadi2
> I'm not sure why this PR is on hold because of PR 339
> This PR needs cdn credentials and the provision to set cdn credentials are already in the cephci.yaml.template
> The PR is actually blocked on procuring live credentials where we actually consume live contents with pool-id with subscription available for testing. The discussion to procure the same is going on.

Ok then will continue with existing cephci.yaml.template
@VasishtaShastry thanks for the update.

**Comment by VasishtaShastry** [comment]
> @sunilangadi2 Ok then will continue with existing cephci.yaml.template
> @VasishtaShastry thanks for the update.

I'm sorry, though PR 339 is not blocking this PR
Would be optimal if 339 gets merged before this one
As I mentioned the main thing blocking us is getting a CDN credential which VMs can consume

**Comment by psathyan** [comment]
@sunilangadi2 as PR339 is merged, can we work towards closing this PR

**Comment by sunilangadi2** [comment]
> @sunilangadi2 as PR339 is merged, can we work towards closing this PR

Sure, currently testing this PR with new cephci.yaml template.

**Comment by sunilangadi2** [comment]
I will close this PR as it's changes been updated in #555

**Pull request closed without merging**